ts/input/tex/ParseUtil.ts

```diff
@@ -33,668 +33,720 @@ import {entities} from '../../util/Entities.js';
-    return parser.create('node', 'mtext', [], def, textNode);
+    mml = [parser.create('node', 'mstyle', mml, {displaystyle: false, scriptlevel: level})];
+  } else if (mml.length > 1) {
+    // @test Interspersed Text
+    mml = [parser.create('node', 'mrow', mml)];
   }
+  return mml;
+}
 
-  /**
-   * Create an munderover node with the given script position.
-   * @param {TexParser} parser   The current TeX parser.
-   * @param {MmlNode} base       The base node.
-   * @param {MmlNode} script     The under- or over-script.
-   * @param {string} pos         Either 'over' or 'under'.
-   * @param {boolean} stack      True if super- or sub-scripts should stack.
-   * @return {MmlNode}           The generated node (MmlMunderover or TeXAtom)
-   */
-  export function underOver(parser: TexParser, base: MmlNode, script: MmlNode, pos: string, stack: boolean): MmlNode {
-    // @test Overline
-    ParseUtil.checkMovableLimits(base);
-    if (NodeUtil.isType(base, 'munderover') && NodeUtil.isEmbellished(base)) {
-      // @test Overline Limits
-      NodeUtil.setProperties(NodeUtil.getCoreMO(base), {lspace: 0, rspace: 0});
-      const mo = parser.create('node', 'mo', [], {rspace: 0});
-      base = parser.create('node', 'mrow', [mo, base]);
-      // TODO? add an empty <mi> so it's not embellished any more
-    }
-    const mml = parser.create('node', 'munderover', [base]) as MmlMunderover;
-    NodeUtil.setChild(mml, pos === 'over' ?  mml.over : mml.under, script);
-    let node: MmlNode = mml;
-    if (stack) {
-      // @test Overbrace 1 2 3, Underbrace, Overbrace Op 1 2
-      node = parser.create('node', 'TeXAtom', [mml], {texClass: TEXCLASS.OP, movesupsub: true});
-    }
-    NodeUtil.setProperty(node, 'subsupOK', true);
-    return node;
+
+/**
+ * Parses text internal to boxes or labels.
+ * @param {TexParser} parser The current tex parser.
+ * @param {string} text The text to parse.
+ * @param {EnvList} def The attributes of the text node.
+ * @return {MmlNode} The text node.
+ */
+export function internalText(parser: TexParser, text: string, def: EnvList): MmlNode {
+  // @test Label, Fbox, Hbox
+  text = text.replace(/\n+/g, ' ').replace(/^\s+/, entities.nbsp).replace(/\s+$/, entities.nbsp);
+  let textNode = parser.create('text', text);
+  return parser.create('node', 'mtext', [], def, textNode);
+}
+
+/**
+ * Create an munderover node with the given script position.
+ * @param {TexParser} parser   The current TeX parser.
+ * @param {MmlNode} base       The base node.
+ * @param {MmlNode} script     The under- or over-script.
+ * @param {string} pos         Either 'over' or 'under'.
+ * @param {boolean} stack      True if super- or sub-scripts should stack.
+ * @return {MmlNode}           The generated node (MmlMunderover or TeXAtom)
+ */
+export function underOver(parser: TexParser, base: MmlNode, script: MmlNode, pos: string, stack: boolean): MmlNode {
+  // @test Overline
+  checkMovableLimits(base);
```

Comment:
> This should remain `ParseUtil.checkMovableLimits(base);` so that if that method is overridden, it will affect this method.
> 
> Ideally, these functions would **not** be exported except through the `ParseUtil` object, so that all usage of the functions are through `ParseUtil`. If you allows for `import {underOver} from 'ParseUtil.js'` that means that `underOver` can't be properly overridden, because the function with the `import` will not get the overridden version.
> 

That's what I meant to emulate with the object at the end of the file. I just wanted to get feedback first before rewriting all the methods into the structure you indicate below.


> I had meant to have the original `namespace ParseUtil {` be replaced by `export const ParseUtil = {`, and all the constants and functions would be part of this object directory. So you would do
> 
> ```js
> export const ParseUtil = {
> 
>   /**
>    * Matches for a dimension argument.
>    * @param {string} dim The argument.
>    * @param {boolean} rest Allow for trailing garbage in the dimension string.
>    * @return {[string, string, number]} The match result as (Anglosaxon) value,
>    *     unit name, length of matched string. The latter is interesting in the
>    *     case of trailing garbage.
>    */
>   matchDimen(dim: string, rest: boolean = false): [string, string, number] {
>     let match = dim.match(rest ? UNIT_CASES.dimenRest : UNIT_CASES.dimenEnd);
>     return match ?
>       this.muReplace([match[1].replace(/,/, '.'), match[4], match[0].length]) :
>       [null, null, 0];
>   },
> 
>   /**
>    * Transforms mu dimension to em if necessary.
>    * @param {[string, string, number]} [value, unit, length] The dimension triple.
>    * @return {[string, string, number]} [value, unit, length] The transformed triple.
>    */
>   muReplace([value, unit, length]: [string, string, number]): [string, string, number] {
>     if (unit !== 'mu') {
>       return [value, unit, length];
>     }
>     let em = this.Em(UNIT_CASES.get(unit) * (parseFloat(value || '1')));
>     return [em.slice(0, -2), 'em', length];
>   },
> 
>   .
>   .
>   .
> 
> };
> ```
> 
> You could also put `UNIT_CASES` as a property as well, rather than exporting it separately, and use `this.UNIT_CASES` within the functions.

Well, ultimately that will move into lengths (together with the methods under `// Temporary before move to lenghts`). But we can do that for now.